### DISCUSSION TITLE
Possible new configuration?

### DISCUSSION THREAD

**Discussion by MrAmericanMike** [tauri-apps/tauri#4864] [General]
So maybe there is a reason that I am missing, but honestly, I see no point as to why this is part of the code:

```rs
#![cfg_attr(
    all(not(debug_assertions), target_os = "windows"),
    windows_subsystem = "windows"
)]
```

Couldn't this be moved to be a config value inside `tauri.conf.json`?

It would clean the main code by removing what looks like plain boilerplate.

**Comment by FabianLars** [comment]
No this can't be moved. It must be at the top (Line 1) of your main.rs file. This is a rust "inconvenience" and it's actually really important, otherwise your app would always open a separate console window on Windows.

Edit: It's a crate attribute (which is why it needs to be at the top), and here's a really small doc about the attribute itself https://doc.rust-lang.org/reference/runtime.html#the-windows_subsystem-attribute
